### ISSUE TITLE
Build system issues

### ISSUE THREAD

**Issue by neusdan** [randombit/botan#828]
I am executing configure.py as follows:

```
configure.py --module-policy bsi --enable-modules=tls,pkcs11,xts
```

Generated output:

```
INFO: configure.py invoked with options "--module-policy bsi --enable-modules=tls,pkcs11,xts"
   INFO: Platform: OS="Windows" machine="AMD64" proc="Intel64 Family 6 Model 94 Stepping 3, GenuineIntel"
   INFO: Guessing target OS is windows (use --os to set)
   INFO: Guessing to use compiler msvc (use --cc to set)
   INFO: Guessing target processor is a x86_64/x86_64 (use --cpu to set)
   INFO: Target is msvc-windows-x86_64-x86_64
   INFO: Skipping, incompatible OS - darwin_secrandom dev_random fd_unix proc_walk
   INFO: Skipping, not requested - adler32 aont bcrypt certstor_sql certstor_sqlite3 chacha_sse2 codec_filt compression crc24 crc32 cryptobox eme_raw ffi filters fpe_fe1 http_util keccak newhope passhash9 pbkdf1 rfc3394 sessions_sql sessions_sqlite3 shake_cipher srp6 tls tls_cbc tss xts
   INFO: Skipping, prohibited by module policy - blake2 blowfish camellia cascade cast cbc_mac ccm cecpq1 cfb chacha chacha20poly1305 comb4p curve25519 des eax elgamal emsa_raw emsa_x931 gost_28147 gost_3410 gost_3411 hkdf idea idea_sse2 kasumi kdf1 kdf2 lion mce mceies md4 misty1 noekeon noekeon_simd ocb ofb poly1305 prf_x942 rc4 rfc6979 rmd160 salsa20 seed serpent serpent_simd shake siphash siv skein threefish threefish_avx2 tiger twofish whirlpool x919_mac xtea
   INFO: Skipping, requires external dependency - boost bzip2 lzma openssl pkcs11 sqlite3 tpm zlib
   INFO: Loading modules aead aes aes_ni aes_ssse3 asn1 auto_rng base base64 bigint block cbc clmul cmac cryptoapi_rng ctr dh dl_algo dl_group dlies dsa dyn_load ec_gfp ec_group ecc_key ecdh ecdsa ecgdsa ecies eckcdsa eme_oaep eme_pkcs1 emsa1 emsa_pkcs1 emsa_pssr entropy gcm gmac hash hash_id hex hmac hmac_drbg iso9796 kdf kdf1_iso18033 keypair locking_allocator mac md5 mdx_hash mgf1 mode_pad modes mp numbertheory par_hash pbes2 pbkdf pbkdf2 pem pk_pad pkcs11 prf_tls pubkey rdrand rdrand_rng rdseed rng rsa sha1 sha1_sse2 sha2_32 sha2_64 sha3 simd sp800_108 sp800_56c stateful_rng stream system_rng tls utils win32_stats x509 xmss xts
   INFO: Enabling use of external dependency pkcs11
   INFO: Assuming CPU is little endian
   INFO: Assuming unaligned memory access works
   INFO: Using hardlink to link files into build dir (use --link-method to change)
   INFO: Botan 2.0.0 (VC git:e500b11cbd47f19c2cb0ead250c427a870ebbfef) (unreleased undated) build setup is complete
```

The modules `tls` and `xts` are requested via `--enable-modules` but the output shows that they are not loaded and listed in the `Skipping, not requested` section.

The `pkcs11` module is loaded, but it shouldn't be listed in `Skipping, requires external dependency`.

**Comment by randombit** [comment]
Thanks, I bet there is some bad interaction between how policies are applied and the dependency logic. Will look

**Comment by neusdan** [comment]
Actually, `tls` and `xts` are loaded and compiled. But they still should not be listed in the `Skipping, not requested` section

**Comment by randombit** [comment]
This works with master. I didn't bisect it but it looks to have been fixed sometime between 2.1 and 2.2. Closing.
